# **HEART+DISEASE**

A. Utiliza un Jupyter Notebook y el paquete Pandas para abrir el archivo y
presentarlo en formato DataFrame, donde el nombre de las columnas
debe corresponder con el nombre real de las variables.

In [1]:
import pandas as pd

In [2]:
columnas = [
    'age', 'sex', 'cp', 'trestbps', 'chol', 'fbs',
    'restecg', 'thalach', 'exang', 'oldpeak', 'slope',
    'ca', 'thal', 'num'
]

In [3]:
archivo = R'processed.cleveland.data'

In [4]:
df = pd.read_csv(archivo, names=columnas)

In [5]:
df #se evidencian datos con el simbolo ?, simbolo faltante

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
298  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
299  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
300  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
301  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  num  
0      3.0  0.0  6.0    0  
1      2.0  3.0  3.0    2  
2      2.0  2.0  7.0    1  
3      3.0  0.0  3.0    0  
4      1.0  0.0  3.0    0  
..     ...  ...  ...  ...  
298    2.0  0.0  7.0    1  
299    2.0  2.0  7.0    2  
300    2.0  1.0  7.0    3  
301    2.0  1.0  3.0    1  
302    1.0    ?  3.0    0  

[303 rows x 14 columns]

B. Analizando el DataFrame, señala las variables numéricas. Entre las
variables numéricas señala cuáles son continuas y cuáles discretas

**Variables Numercias discretas**

age, trestbps, chol, thalach,ca, num

**Variables Numercias continuas**

oldpeak

C. Analizando el DataFrame, señala las variables categóricas. Entre las
variables categóricas señala cuáles son ordinales y cuáles nominales.

**Variables categoricas Ordinales**
fbs, exang


**Variables categoricas nominales**
sex, cp, resecg, slope, thal


D. ¿Hay valores faltantes (missing values) en el dataset? ¿Qué símbolo se
utiliza para marcarlos? ¿Se utiliza más de un tipo de símbolo?

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    float64
 1   sex       303 non-null    float64
 2   cp        303 non-null    float64
 3   trestbps  303 non-null    float64
 4   chol      303 non-null    float64
 5   fbs       303 non-null    float64
 6   restecg   303 non-null    float64
 7   thalach   303 non-null    float64
 8   exang     303 non-null    float64
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    float64
 11  ca        303 non-null    object 
 12  thal      303 non-null    object 
 13  num       303 non-null    int64  
dtypes: float64(11), int64(1), object(2)
memory usage: 33.3+ KB


In [7]:
(df == '?').sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          4
thal        2
num         0
dtype: int64

**Valores Faltantes**
Se evidencia valores faltantes en las columnas **ca** y **thal**, se evidencia con el simbolo **?**, no se evidencia mas simbolos para los datos faltantes en este Dataset 

Utilizando métodos de Python, obtén la lista de valores posibles que
puede tomar la variable “thal”. ¿Se puede ordenar estos valores en una
escala de mayor a menor? ¿Podríamos considerarlos valores categóricos
ordinales?

In [8]:
df['thal'].value_counts() #al tener valores faltantes los reemplazare con el dato que mas se repite 

3.0    166
7.0    117
6.0     18
?        2
Name: thal, dtype: int64

In [9]:
moda_thal = df['thal'].mode()[0]
df['thal'].fillna(moda_thal, inplace=True)

In [10]:
df["thal"].unique() #Se evidencia que ya no hay datos faltantes 

array(['6.0', '3.0', '7.0', '?'], dtype=object)

normal(3), defecto fijo(6), defecto reversible(7)

* Normal implica ausencia de anomalía.

* Defecto fijo es más grave: indica daño permanente.

* Defecto reversible sugiere un flujo sanguíneo restringido que puede recuperarse, lo cual es crítico.

In [11]:
df.sort_values(by='thal', ascending=False) #Ordena los datos de thal de mayor a menor 

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
266  52.0  1.0  4.0     128.0  204.0  1.0      0.0    156.0    1.0      1.0   
87   53.0  0.0  3.0     128.0  216.0  0.0      2.0    115.0    0.0      0.0   
248  52.0  1.0  4.0     125.0  212.0  0.0      0.0    168.0    0.0      1.0   
92   62.0  1.0  3.0     130.0  231.0  0.0      0.0    146.0    0.0      1.8   
213  66.0  0.0  4.0     178.0  228.0  1.0      0.0    165.0    1.0      1.0   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
135  55.0  0.0  2.0     135.0  250.0  0.0      2.0    161.0    0.0      1.4   
139  51.0  1.0  3.0     125.0  245.0  1.0      2.0    166.0    0.0      2.4   
140  59.0  1.0  2.0     140.0  221.0  0.0      0.0    164.0    1.0      0.0   
142  52.0  1.0  2.0     128.0  205.0  1.0      0.0    184.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  num  
266    2.0  0.0    ?    2  
87     1.0  0.0    ?    0  
248    1.0  2.0  7.0    3  
92     2.0  3.0  7.0    0  
213    2.0  2.0  7.0    3  
..     ...  ...  ...  ...  
135    2.0  0.0  3.0    0  
139    2.0  0.0  3.0    0  
140    1.0  0.0  3.0    0  
142    1.0  0.0  3.0    0  
302    1.0    ?  3.0    0  

[303 rows x 14 columns]

los valores de thal pueden ordenarse de menor a mayor severidad.
se pueden tratar como categóricos ordinales, especialmente útil si se van a usar modelos como regresiones lineales o árboles

## CENSUS INCOME ##

a. Utiliza un Jupyter Notebook y el paquete Pandas para abrir el archivo y
presentarlo en formato DataFrame, donde el nombre de las columnas
debe corresponder con el nombre real de las variables. Señala qué
variables del DataFrame presentan valores faltantes (missing values)

In [12]:
archivo_dos = R'adult.data'

In [13]:
columnas_dos = [
    "age", "workclass", "fnlwgt", "education", "education-num",
    "marital-status", "occupation", "relationship", "race", "sex",
    "capital-gain", "capital-loss", "hours-per-week", "native-country", "income"
]


In [14]:
df_dos = pd.read_csv(archivo_dos, names=columnas_dos)
df_dos

age          workclass  fnlwgt    education  education-num  \
0       39          State-gov   77516    Bachelors             13   
1       50   Self-emp-not-inc   83311    Bachelors             13   
2       38            Private  215646      HS-grad              9   
3       53            Private  234721         11th              7   
4       28            Private  338409    Bachelors             13   
...    ...                ...     ...          ...            ...   
32556   27            Private  257302   Assoc-acdm             12   
32557   40            Private  154374      HS-grad              9   
32558   58            Private  151910      HS-grad              9   
32559   22            Private  201490      HS-grad              9   
32560   52       Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation    relationship    race  \
0            Never-married        Adm-clerical   Not-in-family   White   
1       Married-civ-spouse     Exec-managerial         Husband   White   
2                 Divorced   Handlers-cleaners   Not-in-family   White   
3       Married-civ-spouse   Handlers-cleaners         Husband   Black   
4       Married-civ-spouse      Prof-specialty            Wife   Black   
...                    ...                 ...             ...     ...   
32556   Married-civ-spouse        Tech-support            Wife   White   
32557   Married-civ-spouse   Machine-op-inspct         Husband   White   
32558              Widowed        Adm-clerical       Unmarried   White   
32559        Never-married        Adm-clerical       Own-child   White   
32560   Married-civ-spouse     Exec-managerial            Wife   White   

           sex  capital-gain  capital-loss  hours-per-week  native-country  \
0         Male          2174             0              40   United-States   
1         Male             0             0              13   United-States   
2         Male             0             0              40   United-States   
3         Male             0             0              40   United-States   
4       Female             0             0              40            Cuba   
...        ...           ...           ...             ...             ...   
32556   Female             0             0              38   United-States   
32557     Male             0             0              40   United-States   
32558   Female             0             0              40   United-States   
32559     Male             0             0              20   United-States   
32560   Female         15024             0              40   United-States   

       income  
0       <=50K  
1       <=50K  
2       <=50K  
3       <=50K  
4       <=50K  
...       ...  
32556   <=50K  
32557    >50K  
32558   <=50K  
32559   <=50K  
32560    >50K  

[32561 rows x 15 columns]

In [15]:
df_dos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  income          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


In [16]:
df_dos[df_dos['occupation'] == ' ?']

age workclass  fnlwgt      education  education-num  \
27      54         ?  180211   Some-college             10   
61      32         ?  293936        7th-8th              4   
69      25         ?  200681   Some-college             10   
77      67         ?  212759           10th              6   
106     17         ?  304873           10th              6   
...    ...       ...     ...            ...            ...   
32530   35         ?  320084      Bachelors             13   
32531   30         ?   33811      Bachelors             13   
32539   71         ?  287372      Doctorate             16   
32541   41         ?  202822        HS-grad              9   
32542   72         ?  129912        HS-grad              9   

               marital-status occupation    relationship                 race  \
27         Married-civ-spouse          ?         Husband   Asian-Pac-Islander   
61      Married-spouse-absent          ?   Not-in-family                White   
69              Never-married          ?       Own-child                White   
77         Married-civ-spouse          ?         Husband                White   
106             Never-married          ?       Own-child                White   
...                       ...        ...             ...                  ...   
32530      Married-civ-spouse          ?            Wife                White   
32531           Never-married          ?   Not-in-family   Asian-Pac-Islander   
32539      Married-civ-spouse          ?         Husband                White   
32541               Separated          ?   Not-in-family                Black   
32542      Married-civ-spouse          ?         Husband                White   

           sex  capital-gain  capital-loss  hours-per-week  native-country  \
27        Male             0             0              60           South   
61        Male             0             0              40               ?   
69        Male             0             0              40   United-States   
77        Male             0             0               2   United-States   
106     Female         34095             0              32   United-States   
...        ...           ...           ...             ...             ...   
32530   Female             0             0              55   United-States   
32531   Female             0             0              99   United-States   
32539     Male             0             0              10   United-States   
32541   Female             0             0              32   United-States   
32542     Male             0             0              25   United-States   

       income  
27       >50K  
61      <=50K  
69      <=50K  
77      <=50K  
106     <=50K  
...       ...  
32530    >50K  
32531   <=50K  
32539    >50K  
32541   <=50K  
32542   <=50K  

[1843 rows x 15 columns]

Se evidencia que el dataset tienen datos faltantes representados con el signo **?** en las columnas **workclass** y **occupation** al ser variables tipo ***object*** no reconoce el signo como dato faltante 

b. Analizando el DataFrame, señala las variables numéricas y categóricas,
especificando si son continuas, discretas, ordinales o nominales.

**Variables Numericas**

* **Continuas:** fnlwgt, capital-gain, capital-loss, hours-per-week

* **Discretas:** age, education-num, 

**Variables Categoricas**

* **Nominales:** marital-status, occupation, relationship, race, sex

* **Ordinales:** education


c. Utilizando métodos de Python, deduce si hay registros repetidos en el
dataset. ¿Cuántos elementos hay en el DataFrame original? ¿Cuántos
elementos repetidos hay? ¿Cuántos elementos únicos?

In [17]:
df_dos[df_dos.duplicated()]

age          workclass  fnlwgt      education  education-num  \
4881    25            Private  308144      Bachelors             13   
5104    90            Private   52386   Some-college             10   
9171    21            Private  250051   Some-college             10   
11631   20            Private  107658   Some-college             10   
13084   25            Private  195994        1st-4th              2   
15059   21            Private  243368      Preschool              1   
17040   46            Private  173243        HS-grad              9   
18555   30            Private  144593        HS-grad              9   
18698   19            Private   97261        HS-grad              9   
21318   19            Private  138153   Some-college             10   
21490   19            Private  146679   Some-college             10   
21875   49            Private   31267        7th-8th              4   
22300   25            Private  195994        1st-4th              2   
22367   44            Private  367749      Bachelors             13   
22494   49   Self-emp-not-inc   43479   Some-college             10   
25872   23            Private  240137        5th-6th              3   
26313   28            Private  274679        Masters             14   
28230   27            Private  255582        HS-grad              9   
28522   42            Private  204235   Some-college             10   
28846   39            Private   30916        HS-grad              9   
29157   38            Private  207202        HS-grad              9   
30845   46            Private  133616   Some-college             10   
31993   19            Private  251579   Some-college             10   
32404   35            Private  379959        HS-grad              9   

            marital-status          occupation    relationship  \
4881         Never-married        Craft-repair   Not-in-family   
5104         Never-married       Other-service   Not-in-family   
9171         Never-married      Prof-specialty       Own-child   
11631        Never-married        Tech-support   Not-in-family   
13084        Never-married     Priv-house-serv   Not-in-family   
15059        Never-married     Farming-fishing   Not-in-family   
17040   Married-civ-spouse        Craft-repair         Husband   
18555        Never-married       Other-service   Not-in-family   
18698        Never-married     Farming-fishing   Not-in-family   
21318        Never-married        Adm-clerical       Own-child   
21490        Never-married     Exec-managerial       Own-child   
21875   Married-civ-spouse        Craft-repair         Husband   
22300        Never-married     Priv-house-serv   Not-in-family   
22367        Never-married      Prof-specialty   Not-in-family   
22494   Married-civ-spouse        Craft-repair         Husband   
25872        Never-married   Handlers-cleaners   Not-in-family   
26313        Never-married      Prof-specialty   Not-in-family   
28230        Never-married   Machine-op-inspct   Not-in-family   
28522   Married-civ-spouse      Prof-specialty         Husband   
28846   Married-civ-spouse        Craft-repair         Husband   
29157   Married-civ-spouse   Machine-op-inspct         Husband   
30845             Divorced        Adm-clerical       Unmarried   
31993        Never-married       Other-service       Own-child   
32404             Divorced       Other-service   Not-in-family   

                      race      sex  capital-gain  capital-loss  \
4881                 White     Male             0             0   
5104    Asian-Pac-Islander     Male             0             0   
9171                 White   Female             0             0   
11631                White   Female             0             0   
13084                White   Female             0             0   
15059                White     Male             0             0   
17040                White     Male             0             0   
18555                Black     Male         

***¿Cuántos elementos hay en el DataFrame original?***
hay  un total de 32561 filas y 15 columnas

***¿Cuántos elementos repetidos hay?***
hay 24 elementos repetidos

***¿Cuántos elementos únicos?***
hay 32540 elementos unicos

In [18]:
df_dos.nunique()

age                  73
workclass             9
fnlwgt            21648
education            16
education-num        16
marital-status        7
occupation           15
relationship          6
race                  5
sex                   2
capital-gain        119
capital-loss         92
hours-per-week       94
native-country       42
income                2
dtype: int64

***¿Cuántos elementos únicos?***
se evidencia que al momento de consultar los datos unicos estos se ven por columna


d. Utilizando métodos de Python, obtén un DataFrame llamado df_dup en
el que estén los elementos repetidos del DataFrame original. ¿Cuántos de
ellos tienen un país de origen indeterminado? ¿Cuál es la proporción
entre mujeres y hombres? ¿Cuántos cobran más de 50 mil dólares?


In [19]:
df_dup = df_dos[df_dos.duplicated()]
df_dup 

age          workclass  fnlwgt      education  education-num  \
4881    25            Private  308144      Bachelors             13   
5104    90            Private   52386   Some-college             10   
9171    21            Private  250051   Some-college             10   
11631   20            Private  107658   Some-college             10   
13084   25            Private  195994        1st-4th              2   
15059   21            Private  243368      Preschool              1   
17040   46            Private  173243        HS-grad              9   
18555   30            Private  144593        HS-grad              9   
18698   19            Private   97261        HS-grad              9   
21318   19            Private  138153   Some-college             10   
21490   19            Private  146679   Some-college             10   
21875   49            Private   31267        7th-8th              4   
22300   25            Private  195994        1st-4th              2   
22367   44            Private  367749      Bachelors             13   
22494   49   Self-emp-not-inc   43479   Some-college             10   
25872   23            Private  240137        5th-6th              3   
26313   28            Private  274679        Masters             14   
28230   27            Private  255582        HS-grad              9   
28522   42            Private  204235   Some-college             10   
28846   39            Private   30916        HS-grad              9   
29157   38            Private  207202        HS-grad              9   
30845   46            Private  133616   Some-college             10   
31993   19            Private  251579   Some-college             10   
32404   35            Private  379959        HS-grad              9   

            marital-status          occupation    relationship  \
4881         Never-married        Craft-repair   Not-in-family   
5104         Never-married       Other-service   Not-in-family   
9171         Never-married      Prof-specialty       Own-child   
11631        Never-married        Tech-support   Not-in-family   
13084        Never-married     Priv-house-serv   Not-in-family   
15059        Never-married     Farming-fishing   Not-in-family   
17040   Married-civ-spouse        Craft-repair         Husband   
18555        Never-married       Other-service   Not-in-family   
18698        Never-married     Farming-fishing   Not-in-family   
21318        Never-married        Adm-clerical       Own-child   
21490        Never-married     Exec-managerial       Own-child   
21875   Married-civ-spouse        Craft-repair         Husband   
22300        Never-married     Priv-house-serv   Not-in-family   
22367        Never-married      Prof-specialty   Not-in-family   
22494   Married-civ-spouse        Craft-repair         Husband   
25872        Never-married   Handlers-cleaners   Not-in-family   
26313        Never-married      Prof-specialty   Not-in-family   
28230        Never-married   Machine-op-inspct   Not-in-family   
28522   Married-civ-spouse      Prof-specialty         Husband   
28846   Married-civ-spouse        Craft-repair         Husband   
29157   Married-civ-spouse   Machine-op-inspct         Husband   
30845             Divorced        Adm-clerical       Unmarried   
31993        Never-married       Other-service       Own-child   
32404             Divorced       Other-service   Not-in-family   

                      race      sex  capital-gain  capital-loss  \
4881                 White     Male             0             0   
5104    Asian-Pac-Islander     Male             0             0   
9171                 White   Female             0             0   
11631                White   Female             0             0   
13084                White   Female             0             0   
15059                White     Male             0             0   
17040                White     Male             0             0   
18555                Black     Male         

¿Cuántos de ellos tienen un país de origen indeterminado?
* Se evidencia solo un registro con pais indeterminado ya que falta el dato y se representa con un "?"


In [20]:
df_dup[df_dup['occupation'] == ' Other-service']

age workclass  fnlwgt      education  education-num  marital-status  \
5104    90   Private   52386   Some-college             10   Never-married   
18555   30   Private  144593        HS-grad              9   Never-married   
31993   19   Private  251579   Some-college             10   Never-married   
32404   35   Private  379959        HS-grad              9        Divorced   

           occupation    relationship                 race      sex  \
5104    Other-service   Not-in-family   Asian-Pac-Islander     Male   
18555   Other-service   Not-in-family                Black     Male   
31993   Other-service       Own-child                White     Male   
32404   Other-service   Not-in-family                White   Female   

       capital-gain  capital-loss  hours-per-week  native-country  income  
5104              0             0              35   United-States   <=50K  
18555             0             0              40               ?   <=50K  
31993             0             0              14   United-States   <=50K  
32404             0             0              40   United-States   <=50K

¿Cuántos tienen un tipo de trabajo indeterminado? 
* Se evidencia que hay 4 registros que tienen trabajo indeterminado con el valor "Other-service"

¿Cuántos tienen ambas variables indeterminadas a la vez?
* Solo se evidencia un registro que tiene ambas variables indeterminadas 


In [21]:
df_dup['sex'].value_counts(normalize=True)

 Male      0.625
 Female    0.375
Name: sex, dtype: float64

¿cuál es la proporción entre hombres y mujeres?
* vemos que hay un 62% de mujeres y un 37% de hombres 

In [22]:
(df_dup['income'] == " <=50K").mean()*100

91.66666666666666

¿Qué porcentaje de ellos cobra menos de 50 mil dólares?
* el 91% cobra menos de 50 mil dolares

la variable “fnlwgt”. ¿Qué representa?
* indica cuántas personas representa ese individuo dentro de la población total de EE.UU., según factores como edad, raza, sexo, etc.
* Si una persona tiene un fnlwgt = 1,000, significa que ese individuo representa a unas 1,000 personas con características similares en la población general

¿crees que tiene sentido, desde un punto de vista de Machine Learning, eliminar los registros duplicados en este dataset?
* Si, por lo general se ignora para que no distorsione los resultados, la variable se usa mas para analisis poblacional


¿Qué dificultades has encontrado a la hora de trabajar con los valores de las variables de este dataset?
* al principio notar los valores faltantes, ya que estan en variables tipo texto(Object) bajo el signo "?" por lo cual no los mostraba como valores faltantes con codigo python.

¿Sería apropiado buscar una manera automática, usando métodos de Python, de eliminar esa dificultad o dificultades?
¿Se te ocurre cómo hacerlo?
* Si, existe. Sabiendo como se representan los valores faltantes se puede con un df.replace()



## ACCIDENTES BICICLETA ##

In [23]:
df_tres = pd.read_csv('AccidentesBicicletas_2019.csv', sep=';')
df_tres

num_expediente       fecha      hora  \
0      2019S000036  02/01/2019  20:45:00   
1      2019S000045  03/01/2019  10:30:00   
2      2019S000132  03/01/2019  12:45:00   
3      2019S000132  03/01/2019  12:45:00   
4      2019S000133  03/01/2019  14:30:00   
..             ...         ...       ...   
886    2019S039796  19/12/2019  11:40:00   
887    2019S039799  19/12/2019   9:45:00   
888    2019S039844  30/12/2019  12:05:00   
889    2019S039890  27/12/2019  18:30:00   
890    2019S039890  27/12/2019  18:30:00   

                                        localizacion numero  cod_distrito  \
0    AVDA. GRAN VIA DE HORTALEZA / GTA. LUIS ROSALES    65B            16   
1                        CTRA. DEHESA DE LA VILLA, 1      1             9   
2        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
3        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
4                           CALL. FELIPE ALVAREZ, 10     10            18   
..                                               ...    ...           ...   
886                          CALL. ANGEL GANIVET, 28     28             3   
887                CALL. WALIA / CALL. ANGEL GANIVET     21             3   
888                           CALL. PALAS DE REY, 27     27             8   
889                      CALL. ALEJANDRO SANCHEZ, 12     12            11   
890                      CALL. ALEJANDRO SANCHEZ, 12     12            11   

                distrito           tipo_accidente estado_meteorológico  \
0              HORTALEZA  Colisión fronto-lateral            Despejado   
1        MONCLOA-ARAVACA                    Caída            Despejado   
2      VILLA DE VALLECAS                  Alcance            Despejado   
3      VILLA DE VALLECAS                  Alcance            Despejado   
4      VILLA DE VALLECAS                  Alcance         Se desconoce   
..                   ...                      ...                  ...   
886               RETIRO                    Caída         Lluvia débil   
887               RETIRO                    Caída            Despejado   
888  FUENCARRAL-EL PARDO         Colisión lateral            Despejado   
889          CARABANCHEL      Atropello a persona         Se desconoce   
890          CARABANCHEL      Atropello a persona         Se desconoce   

                         tipo_vehículo tipo_persona       rango_edad    sexo  \
0                            Bicicleta    Conductor  De 70 a 74 años  Hombre   
1                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
2                            Bicicleta    Conductor  De 18 a 20 años  Hombre   
3                            Bicicleta    Conductor  De 21 a 24 años  Hombre   
4                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
..                                 ...          ...              ...     ...   
886                          Bicicleta    Conductor  De 55 a 59 años   Mujer   
887  Bicicleta EPAC (pedaleo asistido)    Conductor  De 40 a 44 años  Hombre   
888                          Bicicleta    Conductor  De 10 a 14 años   Mujer   
889                          Bicicleta    Conductor  De 30 a 34 años  Hombre   
890                          Bicicleta       Peatón  Menor de 5 años  Hombre   

     cod_lesividad                                     tipo_lesividad  \
0              3.0                        Ingreso superior a 24 horas   
1              2.0                Ingreso inferior o igual a 24 horas   
2              7.0  Asistencia sanitaria sólo en el lugar del acci...   
3             14.0                           Sin asistencia sanitaria   
4              7.0  Asistencia sanitaria sólo en el lugar del acci...   
..             ...                                                ...   
886            7.0  Asistencia sanitaria sólo en el lugar del acci...   
887            6.0  Asistencia sanitaria inmediata en centro de sa...   
888            7.0  Asistencia sanitaria só

In [24]:
df_tres.info() #algunas variables numericas es tan en tipo texto 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 19 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   num_expediente        891 non-null    object 
 1   fecha                 891 non-null    object 
 2   hora                  891 non-null    object 
 3   localizacion          891 non-null    object 
 4   numero                891 non-null    object 
 5   cod_distrito          891 non-null    int64  
 6   distrito              891 non-null    object 
 7   tipo_accidente        891 non-null    object 
 8   estado_meteorológico  865 non-null    object 
 9   tipo_vehículo         891 non-null    object 
 10  tipo_persona          891 non-null    object 
 11  rango_edad            891 non-null    object 
 12  sexo                  891 non-null    object 
 13  cod_lesividad         795 non-null    float64
 14  tipo_lesividad        795 non-null    object 
 15  coordenada_x_utm      8

Analizando el DataFrame, señala las variables numéricas y categóricas,
especificando si son continuas, discretas, ordinales o nominales.

**Variables Numericas** *se debe notar algunas variables numericas que estan como tipo texto*

* **Discretas:** num_expediente, cod_distrito, positiva_droga

* **Continuas:** cod_lesividad, cod_lesividad, cordenada_x, cordenada_y

**Variables Categoricas**

* **Nominales:** localizacion, distrito, tipo_accidente, estado_meteorologico, tipo_vehiculo, tipo_persona, sexo, positiva_alcohol, tipo_lesividad

* **Ordinales:** rango_edad, 

In [25]:
df_tres.isnull().sum()

num_expediente            0
fecha                     0
hora                      0
localizacion              0
numero                    0
cod_distrito              0
distrito                  0
tipo_accidente            0
estado_meteorológico     26
tipo_vehículo             0
tipo_persona              0
rango_edad                0
sexo                      0
cod_lesividad            96
tipo_lesividad           96
coordenada_x_utm          0
coordenada_y_utm          0
positiva_alcohol          2
positiva_droga          887
dtype: int64

¿Hay valores faltantes en el dataset? Especifica las variables que presentan valores faltantes y cuántos de ellos hay en cada variable.

Se evidencian 5 variables con valores faltantes los cuales son:
* estado_meteorológico     26
* cod_lesividad            96
* tipo_lesividad           96
* positiva_alcohol          2
* positiva_droga          887


In [26]:
df_tres['tipo_lesividad'].nunique()

8

In [27]:
df_tres['tipo_lesividad'].unique()

array(['Ingreso superior a 24 horas',
       'Ingreso inferior o igual a 24 horas',
       'Asistencia sanitaria sólo en el lugar del accidente',
       'Sin asistencia sanitaria',
       'Atención en urgencias sin posterior ingreso', nan,
       'Asistencia sanitaria inmediata en centro de salud o mutua',
       'Asistencia sanitaria ambulatoria con posterioridad',
       'Fallecido 24 horas'], dtype=object)

¿Cuántos valores distintos toma la variable ‘lesividad’? 

* 8 valores unicos

Especifica esos valores. 

* Se muestran en el codigo anterior

¿Qué tipo de variable es la variable ‘lesividad’?

* categorica nominal ya que no tiene un orden o jerarquia

despues de leer el PDF ¿Qué tipo de variable consideras que puede ser la variable ‘lesividad’?

* despues de leer puede ser una variable categorica ordinal 

¿Tu respuesta en este apartado es distinta a tu respuesta del apartado anterior?

* Si, crei que era una categorica nominal ya que no veia un orden o jerarquia, pero leyendo el PDF si se ve una jerarquia

¿Cómo codificarías esta variable a la hora de utilizar este DataFrame como entrada de un modelo de Machine Learning?

* Se podria usar para predecir la gravedad del accidente o la probabilidad de accidente en cierto distrito 

In [28]:
df_tres['positiva_alcohol'].value_counts(normalize=True)

N    0.987627
S    0.012373
Name: positiva_alcohol, dtype: float64

Sin tener en cuenta los valores faltantes, ¿qué proporción de los accidentes registrados presentaban un test de alcohol positivo?

* Se evidencia que los accidentes que presentaban un test de alcohol positivo solo es 0,012% de los accidentes registrados 